SYCL/Basic/half_builtins.cpp

```diff
@@ -166,33 +166,34 @@ template <int N> bool check(vec<float, N> a, vec<float, N> b) {
 
 int main() {
   queue q;
-  float16 a, b, c, d;
-  for (int i = 0; i < SZ_max; i++) {
-    a[i] = i / (float)SZ_max;
-    b[i] = (SZ_max - i) / (float)SZ_max;
-    c[i] = (float)(3 * i);
-  }
-  int err = 0;
-  {
-    buffer<float16> a_buf(&a, 1);
-    buffer<float16> b_buf(&b, 1);
-    buffer<float16> c_buf(&c, 1);
-    buffer<int> err_buf(&err, 1);
-    q.submit([&](handler &cgh) {
-      auto A = a_buf.get_access<access::mode::read>(cgh);
-      auto B = b_buf.get_access<access::mode::read>(cgh);
-      auto C = c_buf.get_access<access::mode::read>(cgh);
-      auto err = err_buf.get_access<access::mode::write>(cgh);
-      cgh.parallel_for(SZ_max, [=](item<1> index) {
-        size_t i = index.get_id(0);
-        TEST_BUILTIN_1(fabs);
-        TEST_BUILTIN_2(fmin);
-        TEST_BUILTIN_2(fmax);
-        TEST_BUILTIN_3(fma);
+  if (q.get_device().has(sycl::aspect::fp16)) {
```

Comment:
Could this be an early exit instead? Just to avoid the indentation of the following code. Maybe it could also print a warning message just to tell people that the early exit was taken instead of just skipping silently.
Hopefully in the future we should be able to limit the tests to require fp16 support to even be considered.

Reply:
Yes sorry, it is added now.

If I understood correctly the `-fsycl-device-code-split=per_kernel` is still needed for the files that only perform fp16 tests? So that they can be compiled for devices that don't have the fp16 aspect?

Reply:
> If I understood correctly the -fsycl-device-code-split=per_kernel is still needed for the files that only perform fp16 tests? So that they can be compiled for devices that don't have the fp16 aspect?

I don't believe so. As long as no kernels are run by the program unless all the kernels are supported, the test should be fine without the splitting.

Reply:
OK that sounds like it make sense. I'll remove those `-fsycl-device-code-split=per_kernel`.